### PR TITLE
Fix geoserver permission

### PR DESCRIPTION
**Pull request by aashish24**

### CODE REVIEW

**Comment by aashish24** [comment]
@kotfic I believe the problem was related to the fact that geoserver does not work with open jdk. I merged your PR into mine. Can you try these changes on AWS?

Thanks,

**Comment by kotfic** [comment]
@aashish24 running now

**Comment by kotfic** [comment]
@aashish24 I am still seeing the following error from ansible:

```
TASK [dataqs : Django updatelayers] ********************************************
task path: /home/kotfic/kitware/projects/src/dataqs/ansible/roles/dataqs/tasks/main.yml:106
fatal: [52.38.159.242]: FAILED! => {"changed": false, "cmd": "./manage.py updatelayers --settings=geonode.settings", "failed": true, "msg": "stdout: Not enabling BingMaps base layer as a BING_API_KEY is not defined in local_settings.py file.\n[updated] Layer quakes_weekly (1/9)\n\n:stderr: Stopping process because --ignore-errors was not set and an error was found.\nTraceback (most recent call last):\n  File \"./manage.py\", line 28, in <module>\n    execute_from_command_line(sys.argv)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/core/management/__init__.py\", line 399, in execute_from_command_line\n    utility.execute()\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/core/management/__init__.py\", line 392, in execute\n    self.fetch_command(subcommand).run_from_argv(self.argv)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/core/management/base.py\", line 242, in run_from_argv\n    self.execute(*args, **options.__dict__)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/core/management/base.py\", line 285, in execute\n    output = self.handle(*args, **options)\n  File \"/home/ubuntu/geonode/geonode/geoserver/management/commands/updatelayers.py\", line 108, in handle\n    remove_deleted=remove_deleted)\n  File \"/home/ubuntu/geonode/geonode/geoserver/helpers.py\", line 458, in gs_slurp\n    \"bbox_y1\": Decimal(resource.latlon_bbox[3])\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/db/models/manager.py\", line 154, in get_or_create\n    return self.get_queryset().get_or_create(**kwargs)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/modeltranslation/manager.py\", line 341, in get_or_create\n    return super(MultilingualQuerySet, self).get_or_create(**kwargs)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/db/models/query.py\", line 383, in get_or_create\n    obj.save(force_insert=True, using=self.db)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/polymorphic/polymorphic_model.py\", line 90, in save\n    return super(PolymorphicModel, self).save(*args, **kwargs)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/db/models/base.py\", line 545, in save\n    force_update=force_update, update_fields=update_fields)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/db/models/base.py\", line 582, in save_base\n    update_fields=update_fields, raw=raw, using=using)\n  File \"/home/ubuntu/venvs/geonode/local/lib/python2.7/site-packages/django/dispatch/dispatcher.py\", line 185, in send\n    response = receiver(signal=self, sender=sender, **named)\n  File \"/home/ubuntu/geonode/geonode/geoserver/signals.py\", line 395, in geoserver_post_save\n    create_thumbnail(instance, thumbnail_remote_url, thumbnail_create_url, ogc_client=http_client)\n  File \"/home/ubuntu/geonode/geonode/layers/utils.py\", line 686, in create_thumbnail\n    instance.save_thumbnail(filename, image=image)\n  File \"/home/ubuntu/geonode/geonode/base/models.py\", line 562, in save_thumbnail\n    os.makedirs(upload_path)\n  File \"/home/ubuntu/venvs/geonode/lib/python2.7/os.py\", line 157, in makedirs\n    mkdir(name, mode)\nException: ('Failed to process wqp_api_ph_map', OSError(13, 'Permission denied'))\n", "path": "/home/ubuntu/venvs/geonode/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games", "state": "absent", "syspath": ["/home/ubuntu/.ansible/tmp/ansible-tmp-1461010774.09-53610464023936", "/usr/lib/python2.7", "/usr/lib/python2.7/plat-x86_64-linux-gnu", "/usr/lib/python2.7/lib-tk", "/usr/lib/python2.7/lib-old", "/usr/lib/python2.7/lib-dynload", "/usr/local/lib/python2.7/dist-packages", "/usr/lib/python2.7/dist-packages", "/usr/lib/python2.7/dist-packages/PILcompat"]}
```

It appears the hifld layer was created:

```
TASK [dataqs : Create the hifld layers] ****************************************
task path: /home/kotfic/kitware/projects/src/dataqs/ansible/roles/dataqs/tasks/main.yml:101
changed: [52.38.159.242] => {"changed": true, "cmd": ["/home/ubuntu/venvs/geonode/bin/python", "/home/ubuntu/venvs/geonode/src/dataqs/dataqs/hifld/hifld.py"], "delta": "0:02:46.955951", "end": "2016-04-18 20:19:34.156511", "rc": 0, "start": "2016-04-18 20:16:47.200560", "stderr": "Stopping process because --ignore-errors was not set and an error was found.\nNo handlers could be found for logger \"dataqs.processors\"\nStopping process because --ignore-errors was not set and an error was found.\nStopping process because --ignore-errors was not set and an error was found.\nStopping process because --ignore-errors was not set and an error was found.\nStopping process because --ignore-errors was not set and an error was found.\nStopping process because --ignore-errors was not set and an error was found.", "stdout": "Not enabling BingMaps base layer as a BING_API_KEY is not defined in local_settings.py file.", "stdout_lines": ["Not enabling BingMaps base layer as a BING_API_KEY is not defined in local_settings.py file."], "warnings": []}
```

In any case the server does not appear to be available at /geoserver

**Comment by aashish24** [comment]
@kotfic thanks. Make sure that you remove postgres, tomcat, etc. before you try my patch. When I did from a clean state, my error was gone. Also, I did it manually as well. We can try that tomorrow. That wil confirm if it is indeed related to java.

**Comment by kotfic** [comment]
@aashish24 Still having trouble with http://localhost:8080/geoserver -  when i curl this from the server I am getting no response (so no error,  just no content).  I am not seeing errors in `/var/lib/tomcat7/webapps/geoserver/data/logs/geoserver.log` or in any of the logs in `/var/log/tomcat7/*.log`  Is it possible we can get matt to weigh in on this?  I don't know much about debugging tomcat

File: ansible/roles/geonode/tasks/main.yml [hunk 1]
```diff
@@ -151,8 +151,9 @@
   file:
     state: directory
     owner: www-data
-    group: www-data
+    group: "{{ deploy_user }}"
```
Comment (aashish24) [inline ansible/roles/geonode/tasks/main.yml:154]: Does group www-data exists?

**Comment by aashish24** [comment]
LGTM :+1: I am merging it now. Thanks @kotfic

**Pull request merged**